src/migrations/weblinks-store/5.1.4.ts

```diff
@@ -7,19 +7,26 @@ import { docsUrl, slackUrl } from "../../common/vars";
 import type { WeblinkData } from "../../common/weblink-store";
 import type { MigrationDeclaration } from "../helpers";
 
+export const lensWebsiteLinkName = "Lens Website";
+export const lensDocumentationWeblinkName = "Lens Documentation";
+export const lensSlackWeblinkName = "Lens Community Slack";
+export const lensTwitterWeblinkName = "Lens on Twitter";
+export const lensBlogWeblinkName = "Lens Official Blog";
+export const kubernetesDocumentationWeblinkName = "Kubernetes Documentation";
+
 export default {
   version: "5.1.4",
   run(store) {
     const weblinksRaw: any = store.get("weblinks");
     const weblinks = (Array.isArray(weblinksRaw) ? weblinksRaw : []) as WeblinkData[];
 
     weblinks.push(
-      { id: "https://k8slens.dev", name: "Lens Website", url: "https://k8slens.dev" },
-      { id: docsUrl, name: "Lens Documentation", url: docsUrl },
-      { id: slackUrl, name: "Lens Community Slack", url: slackUrl },
-      { id: "https://kubernetes.io/docs/home/", name: "Kubernetes Documentation", url: "https://kubernetes.io/docs/home/" },
-      { id: "https://twitter.com/k8slens", name: "Lens on Twitter", url: "https://twitter.com/k8slens" },
-      { id: "https://medium.com/k8slens", name: "Lens Official Blog", url: "https://medium.com/k8slens" },
+      { id: "https://k8slens.dev", name: lensWebsiteLinkName, url: "https://k8slens.dev" },
+      { id: docsUrl, name: lensDocumentationWeblinkName, url: docsUrl },
+      { id: slackUrl, name: lensSlackWeblinkName, url: slackUrl },
+      { id: "https://twitter.com/k8slens", name: lensTwitterWeblinkName, url: "https://twitter.com/k8slens" },
+      { id: "https://medium.com/k8slens", name: lensBlogWeblinkName, url: "https://medium.com/k8slens" },
+      { id: "https://kubernetes.io/docs/home/", name: kubernetesDocumentationWeblinkName, url: "https://kubernetes.io/docs/home/" },
```

Comment:
the type `Weblink` has the following definition:

```ts
interface Weblink {
  id: string;
  name: string;
  url: string;
}
```

And in most cases (namely when a user adds a new weblink) the `id` will be a `uuid.v4`.

When these were added, whoever added them didn't do that and instead used the URLs as the IDs.

The change in this file is to have it so that the names of all of the default weblinks are constants declared in `vars.ts`. (Users cannot edit the names of weblinks).

---

As for why this is all necessary. Users are able to delete weblinks, even the default ones. So the other migrations are present to change the IDs (to some constants which are not UUIDs but also not URLs). And then the third one is run on every new version to make sure that the stored weblink is the current value from `vars.ts`.

---

Yes this is pretty complex, however, given that users can delete weblinks, and that I didn't want to have to keep all previous slack links in the repo, this is how I think it needs to be done.